admin/class-parcelpro-shipping.php

```diff
@@ -216,6 +216,32 @@ public function calculate_shipping($package = array())
             }
 
             foreach ($this->services as $carrier_name => $carrier) {
+                $lowercaseCarrier  = strtolower($carrier_name);
+                $shipping_time = new DateTimeImmutable();
+                $rawCutoffTime = $this->get_option($lowercaseCarrier . '_last_shipping_time');
+                $isBeforeCutoffTime = $this->isBeforeLastShippingTime($rawCutoffTime);
+
+                if (!$isBeforeCutoffTime) {
+                    $shipping_time = $shipping_time->add(new DateInterval('P1D'));
+                }
+                // Fetch expected delivery day
+                $is_enabled = $this->get_option(
+                    $lowercaseCarrier . '_show_delivery_date'
+                );
+                $delivery_expected = new DateTimeImmutable();
+                if ($is_enabled === 'yes') {
+                    $delivery_expected = $this->api->getDeliveryDate(
+                        $carrier_name,
+                        $shipping_time,
+                        $package['destination']['postcode']
+                    );
+                }
+
+                $formattedDeliveryDate = '';
+                if ($delivery_expected && $is_enabled === 'yes') {
+                    $formattedDeliveryDate = ' (' . $this->formatDeliveryDate($delivery_expected) . ')';
+                }
```

Comment:
Suggestie:

```suggestion
                $formattedDeliveryDate = '';
                if ($is_enabled === 'yes') {
                    $delivery_expected = $this->api->getDeliveryDate(
                        $carrier_name,
                        $shipping_time,
                        $package['destination']['postcode']
                    );
                    
                    if ($delivery_expected) {
                        $formattedDeliveryDate = ' (' . $this->formatDeliveryDate($delivery_expected) . ')';
                    }
                }
```

Ja, zo wordt het een nested if, maar in de huidige flow voelt het wel vreemd om een `delivery_expected` te hebben die default "vandaag" is (wat niet klopt), en heb je op 2 plekken de `is_enabled` check nodig.